Dockerfile

```diff
@@ -1,38 +1,52 @@
-ARG VERSION=unspecified
+ARG GIT_COMMIT=unspecified
+ARG GIT_REMOTE=unspecified
+ARG VERSION=0.0.1
 
-FROM python:3.9.6-alpine
+FROM python:3.9.6
 
+ARG GIT_COMMIT
+ARG GIT_REMOTE
 ARG VERSION
 
 # For a list of pre-defined annotation keys and value types see:
 # https://github.com/opencontainers/image-spec/blob/master/annotations.md
 # Note: Additional labels are added by the build workflow.
+LABEL git_commit=$GIT_COMMIT
+LABEL git_remote=$GIT_REMOTE
 LABEL org.opencontainers.image.authors="mark.feldhousen@cisa.dhs.gov"
 LABEL org.opencontainers.image.vendor="Cybersecurity and Infrastructure Security Agency"
+LABEL version=$VERSION
 
 ARG CISA_UID=421
 ENV CISA_HOME="/home/cisa"
+ENV PCA_REPORT_TOOLS_SRC="/usr/src/pca-report-tools"
 ENV ECHO_MESSAGE="Hello World from Dockerfile"
 
 RUN addgroup --system --gid ${CISA_UID} cisa \
   && adduser --system --uid ${CISA_UID} --ingroup cisa cisa
 
-RUN apk --update --no-cache add \
-ca-certificates \
-openssl \
-py-pip
+RUN apt-get install wget
 
-WORKDIR ${CISA_HOME}
+RUN apt-get update && \
+ apt-get install --no-install-recommends -y texlive texlive-xetex texlive-bibtex-extra
 
-RUN wget -O sourcecode.tgz https://github.com/cisagov/skeleton-python-library/archive/v${VERSION}.tar.gz && \
+VOLUME $CISA_HOME
+
+COPY src/version.txt /src
+
+WORKDIR $PCA_REPORT_TOOLS_SRC
+
+RUN wget -O sourcecode.tgz https://github.com/cisagov/pca-report-library/archive/v${VERSION}.tar.gz && \
   tar xzf sourcecode.tgz --strip-components=1 && \
-  pip install --requirement requirements.txt && \
```

Comment:
The requirements get installed with "CMD ["getenv"]" command at the end. The extra pip install is not needed.

Reply:
You should consolidate your `RUN` commands into a [single line for efficiency](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers) (fewer image layers).

I would take the code from here:
https://github.com/cisagov/pca-report-generator-docker/blob/4d038c2f7b9eab871cc0906b28694563547d5a7c/Dockerfile#L30-L39

And consolidate it into a single `RUN` line like this:
```
RUN wget -O sourcecode.tgz https://github.com/cisagov/pca-report-library/archive/v${VERSION}.tar.gz && \
  tar xzf sourcecode.tgz --strip-components=1 && \
  pip install --requirement requirements.txt && \
  cp -r src/pca_report_library/assets/fonts /usr/share/fonts/truetype/ncats && \
  fc-cache -fsv && \
  chmod +x ${PCA_REPORT_TOOLS_SRC}/var/getenv && \
  ln -snf ${PCA_REPORT_TOOLS_SRC}/var/getenv /usr/local/bin && \
  rm sourcecode.tgz
```

Reply:
Since it came up in a different PR I was just looking at, here's an even cleaner way to do this:
```
RUN pip install --no-cache-dir https://github.com/cisagov/pca-report-library/archive/v${VERSION}.tar.gz && \
  cp -r src/pca_report_library/assets/fonts /usr/share/fonts/truetype/ncats && \
  fc-cache -fsv && \
  chmod +x ${PCA_REPORT_TOOLS_SRC}/var/getenv && \
  ln -snf ${PCA_REPORT_TOOLS_SRC}/var/getenv /usr/local/bin
```

Reply:
Now that the version=unspecified in the Dockerfile, is there a way to specify it for the github link?

<img width="854" alt="Screen Shot 2021-11-12 at 12 59 06 PM" src="https://user-images.githubusercontent.com/19600787/141513180-43995008-5bb6-4651-b5f5-0b270f2d8147.png">

Reply:
That step will work when run by GitHub Actions, but if you want to build the image locally, you can pass in that `ARG` by adding `--build-arg VERSION=0.0.1` (or whatever version you want) to your `docker build` command.  This is mentioned in the README [here](https://github.com/cisagov/pca-report-generator-docker/tree/PCADEV-197-customize#building-from-source).

Reply:
Okay that is working now.

Next I tried consolidating the RUN commands as you mentioned. The first way you have it worked. The second way did not work.

First given way
```
RUN wget -O sourcecode.tgz https://github.com/cisagov/pca-report-library/archive/v${VERSION}.tar.gz && \
  tar xzf sourcecode.tgz --strip-components=1 && \
  pip install --requirement requirements.txt && \
  cp -r src/pca_report_library/assets/fonts /usr/share/fonts/truetype/ncats && \
  fc-cache -fsv && \
  chmod +x ${PCA_REPORT_TOOLS_SRC}/var/getenv && \
  ln -snf ${PCA_REPORT_TOOLS_SRC}/var/getenv /usr/local/bin && \
  rm sourcecode.tgz
```
<img width="846" alt="Screen Shot 2021-11-15 at 12 08 55 PM" src="https://user-images.githubusercontent.com/19600787/141824181-8d74110d-0a29-4e22-b351-7af10f1480c4.png">


Second given way
```
RUN pip install --no-cache-dir https://github.com/cisagov/pca-report-library/archive/v${VERSION}.tar.gz && \
  cp -r src/pca_report_library/assets/fonts /usr/share/fonts/truetype/ncats && \
  fc-cache -fsv && \
  chmod +x ${PCA_REPORT_TOOLS_SRC}/var/getenv && \
  ln -snf ${PCA_REPORT_TOOLS_SRC}/var/getenv /usr/local/bin
```
<img width="849" alt="Screen Shot 2021-11-15 at 12 02 51 PM" src="https://user-images.githubusercontent.com/19600787/141823303-dc2ebbec-1685-4f78-9574-a371252be86a.png">

Reply:
@ameliav - can you please provide the full error output from that `pip install` failure so I can have a better idea of what is going wrong there?

Reply:
The `no-cache-dir` flag only applies to the pip cache, which is not the problem here.  The problem is that we no longer have the `pca-report-library` source code tree at `/usr/src/pca-report-tools` which is needed for the `cp`, `chmod`, and `ln` commands.

In the interest of simplicity here, just go with my first suggestion above that worked.  

Once that is done, I suggest that you rename the `PCA_REPORT_TOOLS_SRC` environment variable to `PCA_REPORT_LIBRARY_SRC` to make it more accurate and less confusing.

Reply:
Resolved in ebc31822cf39304fed84fa7b0e3ecbe8777a113c and a9e9a3efd87bba7b8f0f0ace6a2e6b8db23bf856

Reply:
Looks good!